### DISCUSSION TITLE
Question --ligmask and cmap.

### DISCUSSION THREAD

**Discussion by davidRFB** [RMeli/aescore#28] [Q&A] [answered]
Dear Rocco, 

Thanks for the open code in the repostitory. I am trying to train the model based on the methodology of the [paper](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00536-w). 
However I am not getting the same results on CASF probably because I am confusing some arguments in the train.py script. 

For a model:

When selecting only protein and ligand atoms, the following elements remained (in order of abundance for the ligands, see Additional file [1](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00536-w#MOESM1): Figure S3): H, C, O, N, S, P, F, Cl, Br, I.

The option `--ligmask` should be used right? Because the ligands in  PDBbind2016 only contains these atoms H, C, O, N, S, P, F, Cl, Br, I. the species uses for the NN are going to be only for this 10 atoms. 

For the other model:

When metal centers were kept (see Additional file [1](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00536-w#MOESM1): Figure S4), all atoms outside of the previous list were mapped to a dummy element, X. 
the cmap should be:
```
 -cm '{"X":["Zn","Ca","Mg","Se","Na","Cd","Mn","Fe","Ni","Hg","K","Co","Cu","Li","Cs","Sr"]}'
```
in order to map metals to the dummy atom X which is 0 index in `amap.json`?

Thank you in advance for your attention and help. 

Best,
David

**Comment by RMeli** [comment]
Hi David, sorry for the late reply but I'm AFK. I'll reply to your questions when I'll be back, but in the meantime you could try to dig up this information from the original pipeline available on [Zenodo](https://zenodo.org/records/5076079).

**Comment by davidRFB** [comment]
Hi ! Thanks for pointing the zenodo repo !!  The `--cmap` option was used for mapping the metals to X.  However the `--ligmask` option was not used for training, for the model only containing 10 atoms I will guess the preprocessing of the Heteroatoms took the rest out ?

**Reply by RMeli** [reply, marked as answer]
No worries. I'm sorry it is not well documented, but I'm glad you managed to dig out some information.

> The cmap option was used for mapping the metals to X.

Essentially `--cmap` allows you to map different atoms to a single one (as in the case of metal centers) in order to reduce the number of NNs that need training (especially the ones that would see a low number of AEVs). 

You can, for example, use `--cmap` to map all the halogens into a single specie, etc. 

As you mention, in the original work we tried mapping all the metal centers to a generic "metal" type (`X`), or simply ignoring all the protein heteroatoms (resulting in 10 species, as you saw).

> However the Ligmask option was not used for training

The `--ligmask` option was used to do some experiment, and what it does is to use only the AEVs of the ligand, but computed taking into account the protein atoms as well. You can think of it as a middle ground between the ligand-only model mentioned in the paper (where there is no protein), and the full model (where both the ligand AEVs and the protein AEVs are used). With `--ligmask` you have the same AEVs for the ligand that you would have with the full model, but propagate only those within the NNs instead of both protein and ligand AEVs as in the full model. We did not experiment too much with the `--ligmask` approach.
